src/main/java/org/jabref/gui/groups/GroupColorPicker.java

```diff
@@ -0,0 +1,58 @@
+package org.jabref.gui.groups;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import javafx.scene.paint.Color;
+
+import org.jabref.gui.icon.IconTheme;
+
+public class GroupColorPicker {
+    public static boolean useRandom = false;
+    private static final Random RANDOM = new Random();
+    private static final List<Color> TOP_LEVEL_COLORS = new ArrayList<>();
+    private static final int TRIES = 10000;
+
+    // Get most distinct color
+    private static Color getDistinctHue() {
```

Comment:
This method doesn't take the **existing** colors into consideration.

Please adjust the algorithm to use the colors of the existing subgroup (or all the top level groups) and then calculcate THE ONE most distant color.

I would just work with `H`, not `S` and `V`

---

Some inspiration

(read on later)

```java
import java.awt.Color;
import java.util.ArrayList;
import java.util.List;

public class ColorDistance {

    // Find a Hue value that maximizes the sum of distances
    public static float findFurthestHue(List<Color> colors) {
        float maxDistanceSum = -1;
        float furthestHue = 0;

        // Iterate over the Hue space [0, 1)
        for (int h = 0; h < 360; h++) {
            float candidateHue = h / 360.0f;
            double distanceSum = 0;

            for (Color color : colors) {
                float[] currentHSV = new float[3];
                Color.RGBtoHSB(color.getRed(), color.getGreen(), color.getBlue(), currentHSV);

                float dh = Math.abs(candidateHue - currentHSV[0]);
                dh = Math.min(dh, 1.0f - dh);  // Hue is circular

                distanceSum += dh;
            }

            if (distanceSum > maxDistanceSum) {
                maxDistanceSum = distanceSum;
                furthestHue = candidateHue;
            }
        }

        return furthestHue;
    }

    public static void main(String[] args) {
        // Sample list of colors
        List<Color> colors = new ArrayList<>();
        colors.add(new Color(255, 0, 0));    // Red
        colors.add(new Color(0, 255, 0));    // Green
        colors.add(new Color(0, 0, 255));    // Blue
        colors.add(new Color(255, 255, 0));  // Yellow

        // Find a Hue that is furthest away from all colors in the list
        float furthestHue = findFurthestHue(colors);

        // Output the result
        System.out.println("Furthest Hue: " + furthestHue);
    }
}
````

---

Instead of plainly searching, one can use the `Newton-Cotes formula`.

> Applying Newton-Cotes formulas (like the "Trapezoidal Rule" or "Simpson's Rule") used in numerical integration, we can attempt to optimize our search instead of brute-forcing the entire color space.
>
> One approach to find a value that maximizes the sum of distances is to use a heuristic like the median. For circular quantities like Hue, you may want to convert them to a linear space, e.g., by using the sine and cosine values, before finding the average or median.

```
import java.awt.Color;
import java.util.ArrayList;
import java.util.List;

public class ColorDistance {

    public static float findFurthestHue(List<Color> colors) {
        double sumSin = 0;
        double sumCos = 0;

        for (Color color : colors) {
            float[] hsv = new float[3];
            Color.RGBtoHSB(color.getRed(), color.getGreen(), color.getBlue(), hsv);
            double angle = hsv[0] * 2 * Math.PI;
            sumSin += Math.sin(angle);
            sumCos += Math.cos(angle);
        }

        double meanAngle = Math.atan2(-sumSin, -sumCos) + Math.PI;
        float furthestHue = (float) (meanAngle / (2 * Math.PI));

        return furthestHue;
    }

    public static void main(String[] args) {
        // Sample list of colors
        List<Color> colors = new ArrayList<>();
        colors.add(new Color(255, 0, 0));    // Red
        colors.add(new Color(0, 255, 0));    // Green
        colors.add(new Color(0, 0, 255));    // Blue
        colors.add(new Color(255, 255, 0));  // Yellow

        // Find a Hue that is furthest away from all colors in the list
        float furthestHue = findFurthestHue(colors);

        // Output the result
        System.out.println("Furthest Hue: " + furthestHue);
    }
}
```